src/Integration.Vsix/InfoBar/InfoBarManager.cs

```diff
@@ -102,8 +142,8 @@ public void DetachInfoBar(IInfoBar currentInfoBar)
             currentInfoBar.Close();
 
             ThreadHelper.ThrowIfNotOnUIThread();
-            IVsInfoBarHost host;
-            if (TryGetInfoBarHost(wrapper.Frame, out host))
+            IVsInfoBarHost host = wrapper.Host;
```

Comment:
If its a main window gold bar we already know the host, else try get it.

Reply:
This is a bit untidy because the calling code (i.e. this method) has to know to try the host property first, before trying the frame property.

It doesn't matter too much since `PrivateInforBarWrapper` is a private class so it can be only used within this class.

However, it would be easy to make it neater: move the code in lines 144 to 149 into the `PrivateInfroBarWrapper.Close` method i.e. make the private class responsible for clearing up the info bar UI element and the host.

That hides the special cleanup behaviour from the caller. It might also mean that the ``Frame` and `Host` properties in the private class can both be made `private` if nothing else in the parent class uses them.

It shouldn't affect the existing tests since they can't see the `PrivateInfoBarWrapper` class anyway.